### PR TITLE
Allow configuring the separator used in numbering equations

### PR DESCRIPTION
**Pull request by thfanning**
Subject: Sphinx configuration: `math_numsep`

### Feature or Bugfix
- Feature

### Purpose
- Allow users to configure the final separator used when numbering math elements.

### Detail
- When numbering equations with `math_numfig` and `numfig_secnum_depth > 0`, section number are included in the equation number. For example, Eq. 1.2.3 is the third equation in section 1.2.
- A new Sphinx configuration option `math_numsep` allows users to select a different separator between the section number and the equation number.
- For example, `math_numsep='-'` will result in numbering such as Eq. 1.2-3.
- This helps visually distinguish equation number 1.2-3 from section number 1.2.3.
- Minimal code changes.
- Test case included for HTML. LaTeX does its own numbering.

### Relates
- None

### CODE REVIEW

**Comment by AA-Turner** [comment]
I'm hesistant about adding new configuration values. What is the value proposition here, is it a broader request?

A

**Comment by thfanning** [comment]
> I'm hesistant about adding new configuration values. What is the value proposition here, is it a broader request?

Understandable given the large number of existing configuration options. However, there are only 3 options for math (and `math_number_all` is of questionable utility). What we propose, `math_numsep`, has symmetry with `math_numfig` and only takes effect if `numfig_secnum_depth` is positive.

We maintain very technical code documentation that is thousands of pages long and contains over 4000 equations across 16 chapters. Our convention (for decades) is to use Chapter.Section-Equation. Although this is something that can [readily be done in LaTeX](https://tex.stackexchange.com/questions/420087/how-can-i-use-hyphens-in-equation-numbers-but-periods-in-other-numbers), Sphinx is limited in equation number formatting in HTML (other than the `math_eqref_format` which has no bearing on numbering).

This feature would bring parity with LaTeX generation.

Thanks.

File: sphinx/config.py [hunk 1]
```diff
@@ -258,6 +258,7 @@ class Config:
         'math_number_all': _Opt(False, 'env', ()),
         'math_eqref_format': _Opt(None, 'env', frozenset((str,))),
         'math_numfig': _Opt(True, 'env', ()),
+        'math_numsep': _Opt(None, 'env', frozenset((str,))),
```
Comment (AA-Turner) [inline sphinx/config.py]: Probably makes sense to unconditionally use ``math_numsep``, and have a sensible default. If a user sets ``math_numsep`` to ``None`` or ``''``, perhaps issue a warning/error, or silently use ``'.'``.
```suggestion
        'math_numsep': _Opt('.', 'env', frozenset((str,))),
```

**Comment by AA-Turner** [comment]
Please can you also add an entry to CHANGES and documentation in ``doc/usage/configuration``?

A

File: sphinx/config.py (hunk 1 above)
Comment (thfanning) [reply sphinx/config.py]: I've implemented the requested default separator, added to the CHANGES, and updated the documentation.

**Comment by AA-Turner** [comment]
Thanks Thomas!

A

**Pull request merged**